### PR TITLE
fix(security): upgrade .NET runtime to 8.0.21 and resolve NuGet vulnerabilities

### PR DESCRIPTION
**Pull request by Krishna-Chaitanya4**
## Summary
Updates .NET runtime from 8.0.20 to 8.0.21 and updates NuGet packages to address security vulnerabilities.

### .NET Runtime & SDK
- Updated .NET runtime from 8.0.20 to 8.0.21 across all 18 projects
- Updated .NET SDK from 8.0.414 to 8.0.415 in Azure DevOps build configuration
- Updated Docker base images from `mcr.microsoft.com/dotnet/aspnet:8.0.20` to `8.0.21`

### NuGet Package Updates
- **Arcus packages** (compatibility with Arcus.Observability 3.0.0):
  - Arcus.Observability.Telemetry.Core: 2.8.0 → 3.0.0
  - Arcus.WebApi.Logging: 1.7.1 → 2.0.0
  
- **Microsoft.Extensions packages** (resolved dependency conflicts):
  - Microsoft.Extensions.Logging.Console: 8.0.1 → 9.0.10
  - Microsoft.Extensions.Logging.Abstractions: 8.0.2 → 9.0.10
  - Microsoft.Extensions.Configuration.Abstractions: 8.0.0 → 9.0.0
  - Microsoft.Extensions.Configuration.Binder: 8.0.2 → 9.0.0
  
- **Other security updates**:
  - Polly: 8.6.2 → 8.6.4
  - Microsoft.Azure.Kusto.Language: 12.0.0 → 12.2.0
  - System.Drawing.Common: 9.0.9 → 9.0.10
  - System.Security.Cryptography.Pkcs: 9.0.9 → 9.0.10
  - System.Security.Cryptography.Xml: 9.0.9 → 9.0.10
  - Added explicit System.Text.RegularExpressions 4.3.1 reference to fix HIGH severity vulnerability

## Testing
- ✅ Full solution builds successfully (all 18 projects)
- ✅ All 1,539 unit tests pass
- ✅ NuGet package restore completes without errors or warnings
- ✅ No vulnerable packages detected
- ✅ Docker base images updated and verified

## Security
- Resolves HIGH severity vulnerability in System.Text.RegularExpressions (CVE tracked)
- Updates runtime to address .NET 8.0.20 security advisories
- All transitive dependencies updated to secure versions

## Files Modified
- 18 `.csproj` files: Updated `RuntimeFrameworkVersion` to 8.0.21
- 2 `Dockerfile.linux` files: Updated base images to 8.0.21
- `build/azure-devops/variables/build.yml`: Updated SDK version to 8.0.415

### CODE REVIEW

File: src/Promitor.Tests.Unit/Prometheus/LabelTransformerTests.cs [hunk 1]
```diff
@@ -15,11 +15,11 @@ public void TransformLabels_NoTransformationWithDifferentCases_NoTransformationW
             // Arrange
             var inputLabels = new Dictionary<string, string>
```
[the remark this answers is not in the document]
Comment (Krishna-Chaitanya4) [reply src/Promitor.Tests.Unit/Prometheus/LabelTransformerTests.cs]: Hey Tom! You're right - the Bogus-based tests work fine. I encountered a rare edge case where Bogus generated duplicate keys (both generated "Ara"), which caused this error during CI:

System.ArgumentException : An item with the same key has already been added. Key: Ara

However, since this is such a rare occurrence, I agree it's not worth changing the test approach. I've reverted the change. The tests should pass on retry if it happens again. Thanks for the feedback! 👍
